c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -482,6 +484,109 @@ class CachingAllocatorConfig {
   }
 };
 
+// create a custom memory plan
+class MemoryPlanner {
+ private:
+  std::vector<std::vector<AllocFreeEvent>> mem_plan;
+  std::vector<Block*> plan_blk_ptr;
+  std::vector<unsigned int> plan_counter;
+  std::vector<std::map<intptr_t, bool>> allocated_blocks;
+  std::vector<size_t> planning_size;
+  std::vector<unsigned int> non_rec_length;
+
+ public:
+  ska::flat_hash_map<void*, Block*> assigned_planned_blocks;
```

Comment:
curious, what are the reasons to use `ska::flat_has_map` here instead of `std::unordered_map`?